### ISSUE TITLE
Adopt consistent style in field and code descriptions

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/standard#850]
The style and apparent audience (e.g. publisher or user) of descriptions in the schema is inconsistent. Compare the `Tender.submissionMethod` field to the 'unsuccessful' code in `awardStatus`.

* [ ] Update schema descriptions for common field types based on the [description templates](https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#field-and-code-descriptions)
* [ ] Update schema descriptions for other fields based on the [style guide](https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#schema-structure)

Some codelist descriptions are explicitly normative and some are implicitly normative (compare `tag` to `Identifier.scheme`).

* [ ] Update codelist descriptions per [codelist description template](http://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/style_guide/#codelists)

**Comment by jpmckinney** [comment]
This is a bigger project, so moving to 1.2.0 (linked to #826 and #827).

**Comment by ColinMaudry** [comment]
@jpmckinney The description of this issue has not aged well as the style guide has grown significantly and field and code decriptions may have changed since then. Could you please update the issue description with updated examples of what needs fixing?

I assume this is the part of the style guide that is relevant today: https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#field-and-code-descriptions

**Comment by jpmckinney** [comment]
I think the same high-level issues need fixing as before. I'm sure once anyone starts on the issue after reading the style guide that they will easily find problematic field/code descriptions. Finding the issues is half the work :)

**Comment by ColinMaudry** [comment]
Wow the style guide seems to have grown a lot, great job @jpmckinney !

**Comment by ColinMaudry** [comment]
@jpmckinney I have run `ocdskit schema-strict` on the the release-schema.json of dev-1.2, and it seems like many properties are missing. Shall I proceed pushing the changes or there is a reason why these properties were not added?

See https://github.com/open-contracting/standard/commit/815a5ca10f1499cbf4562d70779ea258d00aa41a

**Comment by ColinMaudry** [comment]
(moved discussion about deprecating `Transaction.source` to https://github.com/open-contracting/standard/issues/1304)

**Comment by jpmckinney** [comment]
> @jpmckinney I have run `ocdskit schema-strict` on the the release-schema.json of dev-1.2, and it seems like many properties are missing. Shall I proceed pushing the changes or there is a reason why these properties were not added?

We can't make the schema more strict until 2.0 (though see also #1046).

`ocdskit schema-strict` should never be run on the standard - only on extensions.

**Comment by JachymHercher** [comment]
1) https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#codelists prescribes the following structure for the description of fields with codelists: 

     `<semantics>, using the <name> codelist. See also the <xDetails> field.`

    I've corrected the "using the 'name'" throughout the text in https://github.com/open-contracting/standard/pull/1303/commits/ffc3bbf100d572e6d690cc5cf1a3b570be82f98a (e.g. "from the 'name' codelist" was common). 

2) In all cases, the formulations are actually "using the _open/closed_ <name> codelist", not "using the <name> codelist". I'm not certain about the usefulness of open/closed, so I wanted to double check before removing them. Alternatively, we can change the guide. My reflex would be to remove them to avoid duplication. (Also see https://github.com/open-contracting/standard/issues/1405.)

3) Another bountiful source of inconsistencies is https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#normative-statements which says "Normative keywords should be capitalized where used, per RFC8174.", which I understand as all caps for MUST etc. We currently don't do that anywhere, so I guess that's another major change to be done? However, we should probably first decide to what extent we want to be explicitly normative or not;  and possibly wait until there are very few unmerged PRs as this will likely influence many documents and would cause conflicts. I would propose that descriptions are explicitly normative only when they really need to.

**Comment by jpmckinney** [comment]
This comment is quite longer than the original notification I received :)

Some parts of the style guide were written some time ago, and they can be either not ideal or incorrect.

1. Which do you find more appropriate - "using" or "from"? I don't have a strong opinion.
2. I think it is a useful reminder, since you know that, if it is closed, then you *must* read the codelist. If it is open, you might trust your memory of the codelist, in terms of whether the concept you are mapping is represented there or not. If the schema doesn't indicate whether it is closed, I worry that perhaps more implementers will trust their memory.
3. I prefer to remove the rule about all-caps. It'll create major conflicts, it looks ugly, and it's really more appropriate for technical standards like HTTP or IP rather than mostly-semantic standards like OCDS. That said, all the normative words in the documentation are there intentionally. We did a major review in 1.1.4 to fix them all.

If there are any other inconsistencies, please make a note of them.

**Comment by JachymHercher** [comment]
1. Me neither. I used "using the" because it was in the guide, happy to stick with it.
2. Ok, so let's just change the guide.
3. Gladly agree.

**Comment by jpmckinney** [comment]
2. Done in https://github.com/open-contracting/standard-development-handbook/commit/567a786
3. Done in https://github.com/open-contracting/standard-development-handbook/commit/a95e98bdfec7d734e0b3588825f7afb8cf35d97c

**Comment by JachymHercher** [comment]
4. We should replace "procuring entity" with "buyer or procuring entity". We often use "procuring entity" as buyer (or as an umbrella term) because in UNCITRAL that's what it means, but that's not in line with OCDS descriptions. (We could have done this "clean-up" as a follow-up to https://github.com/open-contracting/standard/issues/571.)

**Comment by jpmckinney** [comment]
Yes, we should clean up the use of procuring entity as proposed in (4).

**Comment by odscjen** [comment]
Re. (4), the following files require clean up:
* docs/guidance/map/milestones.md
* docs/guidance/map/organizational_units.md
* docs/guidance/map/pre-qualification.md
* docs/guidance/build/change_history.md

Is it worth adding these changes to the branch created for #1303 which is now very far behind the latest version or would it be easier to create a new branch and PR?

**Comment by jpmckinney** [comment]
Create a new PR.

As described in the last comment in #1303, that PR will need to be abandoned as it is too much effort to fix conflicts. (Instead, the changes should be described in a spreadsheet, as any PR containing such widespread changes will result in massive conflicts before all review/comments/etc are addressed.)

**Comment by odscjen** [comment]
@jpmckinney Looking specifically at docs/guidance/build/change_history.md most of the examples refer to "the procuring entity", e.g. line 139 "Because of them the procuring entity and the supplier agree to a contract extension." Based on the summary in https://github.com/open-contracting/standard/issues/571#issuecomment-778510316 which states that 'buyer' is to be preferred over 'procuringEntity' should these all be changed to refer instead to "the buyer"? There are no instances in the example json files of the buyer and the procuring entity being different organizations and while the role 'buyer' is included in every example 'procuringEntity' is only included in some (despite all being the same organization).

**Comment by jpmckinney** [comment]
Yes, it would be better for that example to just use a buyer without a procuring entity. However, such changes should wait on #1531, which makes significant changes to that section of the guidance.

**Comment by odscjen** [comment]
ok, I'll wait for that PR to be merged before creating a PR for this issue to save multiple PR's being required

**Comment by jpmckinney** [comment]
We can review this old PR https://github.com/open-contracting/standard/issues/850, but as noted in the last comment we should propose changes in a spreadsheet for easier review and edits - and without the hassle of conflicts.

**Comment by duncandewhurst** [comment]
@jpmckinney is the task here basically to start the review again from the current field descriptions on `1.2-dev`, but document it in a document rather than making changes directly in a PR? Or should we pull in both the `1.2-dev` descriptions and descriptions from the abandoned PR (https://github.com/open-contracting/standard/pull/1303)?

I'm not sure pulling both in will necessarily save any time as we'll have twice as much to read...

Edit: Noting that, depending on the answer to https://github.com/open-contracting/standard/issues/1406#issuecomment-1596288832, I think that issue can be incorporated into this one so that we don't need to go through all the field descriptions twice.

**Comment by jpmckinney** [comment]
Re: #1406, the updated guidance is at https://ocds-standard-development-handbook.readthedocs.io/en/latest/meta/schema_style_guide.html#articles

That said, the issue notes:

> I figure we will need to add more as we review more descriptions.

Re: #1303, I think we'll catch the same issues it identified, so no need to pull in. Some noteworthy changes:

* Some fields had more rework
  * Identifier.scheme
  * Implementation.milestones
* Omit "(Deprecated in 1.1)" or similar from `deprecated.description` and from the field description.
* Remove redundant phrases like "if applicable".
* If using the nouns from a code in a sentence, use normal case (e.g. "approval letter" in Milestone.code).
* American spelling (favour -> favor).

**Comment by duncandewhurst** [comment]
Sounds good. Thanks!

**Comment by odscjen** [comment]
Starting to note the findings of the review [here](https://docs.google.com/spreadsheets/d/1WaN5cdU5UxoCMkq8Od5zlLj68-ckn1lnzWpt_DfzqHk/edit?usp=sharing), I'll update this issue again once there are enough entries to be reviewed

**Comment by odscjen** [comment]
Re. Titles - there's a mix of titles with all words capitalized and those with only the first word capitalized. I'm assuming the latter is the preference from work on extensions but that doesn't appear to be specified in the style guide? @jpmckinney

**Comment by odscjen** [comment]
A general point about the style guide, there's no specific section on titles of fields. I'm finding mixtures of different patterns etc so would suggest the following:

> A title should:
> * Not simply repeat the description.
> * Include the top-level sub-schema name it is a part of, if the field is used in more than one sub-schema.
> * Only capitalize the first word of the title.

Probably other guidance will occur as I work through this

**Comment by odscjen** [comment]
I'm about half way through so if anyone wants to start reviewing [my suggestions](https://docs.google.com/spreadsheets/d/1WaN5cdU5UxoCMkq8Od5zlLj68-ckn1lnzWpt_DfzqHk/edit#gid=0) please do.

@duncandewhurst @jpmckinney 

Other than the points mentioned in previous comments the only new one is that I've just skipped deprecated fields in their entirety so they may still not match the style guide and/or be inconsistent.

**Comment by jpmckinney** [comment]
Can you provide edit access to the sheet?

Makes sense to add a section on field titles to the handbook.

> Only capitalize the first word of the title.

+1

> Not simply repeat the description.

The description is typically written after the title, so this might be a bullet for the description. What's an example?

> Include the top-level sub-schema name it is a part of, if the field is used in more than one sub-schema.

I think status, title, etc. can just be titled Status, Title, instead of Tender status, Award status, etc.

> I've just skipped deprecated fields in their entirety so they may still not match the style guide and/or be inconsistent.

That's fine.

**Comment by odscjen** [comment]
Ah, sorry I thought I had, you should have edit access now.

> The description is typically written after the title, so this might be a bullet for the description. What's an example?

I think what triggered this for me was `award/description` which has title "Description" and description "Award description". But given your response to the next point I don't think this is necessarily an issue anymore.

> I think status, title, etc. can just be titled Status, Title, instead of Tender status, Award status, etc.

OK, there was inconsistency either way with some titles including the parent object name and some not so I think this needs to be included in the style guide.

**Comment by odscjen** [comment]
I'll go through and remove a load of lines from the Excel sheet where I've added the parent field to the title based on the above

**Comment by jpmckinney** [comment]
I think we should probably try to remove, as much as possible, noun phrases like "this tender" or "the tender". A tender in most places means the offer from the supplier – not the opportunity from the buyer.

However, this task is separate from style, so we should cover it in a new issue. That said, let's skip all `tender` fields in the review until the new issue is closed. #1639

**Comment by jpmckinney** [comment]
> Ah, sorry I thought I had, you should have edit access now.

I've started to review, in two new columns.

Feel free to start an issue or PR in the handbook for any new or updated style guidance.

Note to self: I had been checking https://github.com/open-contracting/standard/pull/1303/files just in case there were some useful changes there.

**Comment by jpmckinney** [comment]
Although we aren't editing deprecated fields, let's still remove content like " (Deprecated in 1.1)" from descriptions.

**Comment by odscjen** [comment]
Noting based on the spreadsheet comments that reordering schema elements should be a seperate PR that some of these are already covered by https://github.com/open-contracting/standard/pull/1382 which is on hold to avoid merge conflicts

**Comment by odscjen** [comment]
Created a new sheet to record the fields with elements in the wrong order (seems sensible to record these at the same time as going through all the fields for the descriptions and titles)

**Comment by odscjen** [comment]
`scheme` is described in a number of ways:

* "The **list, register or system** from which the identifier is taken..."
* "The **scheme or codelist** from which the classification code is taken..."
* "...The scheme field is used to indicate the **list or register** from which the identifier is taken...."
* "The **list** from which identifiers for units of measure are taken..."

It seems we should be using consistent language here much as we do with "goods, services and works".

I think the top option was the most recent addition, it comes from `SimplyIdentifier`, @jpmckinney how do you feel about using "list, register or system" as the consistent phrase?

**Comment by jpmckinney** [comment]
Yes, first one sounds good!

**Comment by odscjen** [comment]
Great, I've added those to the spreadsheet too. I think that's all of the field titles and descriptions reviewed now, I've also updated some of the suggested changes based on your initial feedback. I've indicated where I've done that with a new column I.

**Comment by odscjen** [comment]
just a gentle prod @jpmckinney that all the suggested changes are now in the spreadsheet and waiting on your review

**Comment by jpmckinney** [comment]
Yes, it's in my backlog. As this is the longest issue to review, I'll get to it in 2024.

The sheet: https://docs.google.com/spreadsheets/d/1WaN5cdU5UxoCMkq8Od5zlLj68-ckn1lnzWpt_DfzqHk/edit#gid=0

**Comment by odscjen** [comment]
Been reviewing this as there's been quite a few changes to the schema since the spreadsheet was first created. Realised the `id` or `identifier` associated with a `scheme` is described in a number of different ways across objects and it might be good to be consistent here, at least in the first sentence of each.

object | field | description
-- | -- | --
Classification | `id` | The classification code taken from the scheme. Although an integer is allowed, it is recommended to use a string.
Identifier | `id` | The identifier of the organization in the selected scheme. Although an integer is allowed, it is recommended to use a string.
RelatedProcess | `identifier` | The identifier of the related process. If the scheme is 'ocid', this must be an open contracting process identifier (ocid).
Unit | `id` | The identifier from the codelist referenced in the `scheme` field. Check the [unitClassificationScheme](https://standard.open-contracting.org/{{version}}/{{lang}}/schema/codelists/#unit-classification-scheme) codelist for details of how to find and use identifiers from the scheme in use.
SimpleIdentifier | `id` | The identifier taken from the scheme.

I think it's good to explicitly link the value in these fields to the values in the associated `scheme` fields. So as a consistent first sentence how about:

> The identifier taken from the list, register or system referenced in the `scheme` field.

where the agreed common language to refer to a scheme is used as well

**Comment by odscjen** [comment]
Sorry @jpmckinney  I just realised I'd not added any of the codelists to the spreadsheet! It's now a lot longer.

**Comment by odscjen** [comment]
Some more updates have been added to the 1.2-dev branch since this issue was last looked at. I've now updated the spreadsheet to include any new or updated fields and codes.

@jpmckinney would it make sense for @duncandewhurst to do a first pass review of the google sheet checking off the obvious changes and marking the more indepth ones that would require your input to lessen the amount you have to review?

**Comment by jpmckinney** [comment]
I think it'll be better for me to merge the big open PRs, then check their diffs, and continue from there.
